agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/CommandExecutor.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+
+class CommandExecutor {
+
+  private CommandExecutor() {}
+
+  static String execute(ProcessBuilder processBuilder) {
+    IllegalStateException exitException = null;
+    String result = null;
+    try {
+      Process process = processBuilder.start();
+      int exitValue = process.waitFor();
+      exitException =
+          buildExitValueExceptionIfNecessary(processBuilder.command(), exitValue, process);
+      if (exitException == null) {
+        InputStream inputStream = process.getInputStream();
+        result = toString(inputStream);
+      }
+      process.destroy();
+    } catch (IllegalStateException | IOException | InterruptedException e) {
+      throw combineExceptionsIfNecessary(exitException, e, processBuilder.command());
+    }
+    if (exitException != null) {
+      throw exitException;
+    }
+    return result;
+  }
+
+  static String executeWithoutException(ProcessBuilder processBuilder, Logger startupLogger) {
+    try {
+      return execute(processBuilder);
+    } catch (RuntimeException e) {
+      startupLogger.error("Error when executing command " + processBuilder.command() + ".", e);
+      if (e.getSuppressed().length == 1) {
+        return e.getMessage() + " (Suppressed: " + e.getSuppressed()[0] + ")";
+      }
```

Comment:
The suppressed message doesn't seem like it would be very interesting typically, and this would allow still getting the full stack trace along with suppressed exception if really needed at debug level.
```suggestion
      startupLogger.debug(e.getMessage(), e);
```